pycozmo/conn.py

```diff
@@ -110,11 +110,16 @@ def __init__(self,
         self.timeout = timeout
         self.stop_flag = False
         self.queue = Queue()
+        self.to_clear_queue = Queue()
         self.last_ack = 1
         self.last_send_timestamp = None
+        self.one_time_queue = []
+        self.repeat_queue = {}
```

Comment:
Would you be willing to leave the resend changes for a separate PR?

While adding packet resending is very needed to achieve robust communication, it is completely separate from the audio changes.

In general, the `self.window` which is an instance of `SendWindow` acts as both `repeat_queue` and `to_clear_queue`. Implementing resend boils down to resending packets (just the first?) in the current send window after a timeout. This could be done either in a new resend thread or in the current `SendThread` when `self.queue.get()` times out.

Reply:
For sure, that makes total sense. We should be aware though that without this, there will be problems with audio synchronization. It looks like it is important to keep the Keyframe count to know when the packet has been received. Not including this would make the robot miss frames, producing some clipping sounds. I think that both changes are needed for the sound to work properly.

Reply:
Agreed. I am now motivated to get this working :)